ddp-workspace/projects/ddp-sdk/src/lib/components/user/userPreferences.component.ts

```diff
@@ -170,62 +153,62 @@ export class UserPreferencesComponent implements OnDestroy {
             .subscribe((isBusy) => {
                 this.loaded = !isBusy;
             });
-        this.addressSubmitFinalized$
-            .pipe(takeUntil(this.ngUnsubscribe))
-            .subscribe((result) => {
-                this.addressReadonly = false;
-                if (result) {
-                    this.dialogRef.close();
-                }
-            });
     }
 
     public ngOnDestroy(): void {
         this.ngUnsubscribe.next();
         this.ngUnsubscribe.complete();
     }
 
-    public isDateReadyForSave(isFirstInput: boolean): boolean {
-        if (!this.yearOfBirth && !this.monthOfBirth && !this.dayOfBirth) {
-            return true;
-        }
-        if (!this.yearOfBirth || !this.monthOfBirth || !this.dayOfBirth) {
-            return !isFirstInput;
-        }
-        return this.dateService.checkExistingDate(this.yearOfBirth, this.monthOfBirth, this.dayOfBirth);
+    private getBirthDateValidator(): ValidatorFn {
+        return (control: FormControl): ValidationErrors | null => {
+            const value = control.value;
+            if (!value.year || !value.month || !value.day) {
+                return { invalidDate: true };
+            }
+            return this.dateService.checkExistingDate(value.year, value.month, value.day) ? null : { invalidDate: true };
+        };
     }
 
-    public get isDateValid(): boolean {
-        return this.isDateReadyForSave(this.showError);
+    public canSaveUserPreferences(): boolean {
+        return this.loaded && this.isAddressValid
+            // let's allow to save preferences if user didn't touch empty birthday date since s/he can fill it later at action forms
+            && (this.birthDate.valid || (this.birthDate.invalid && this.birthDate.pristine));
     }
 
     public save(): void {
-        if (!this.isDateReadyForSave(true)) {
-            this.showError = true;
+        if (!(this.profileForm.dirty || this.addressDirty)) {
+            this.dialogRef.close();
```

Comment:
Would checking that address component is busy sufficient to decide whether we can close the dialog?